app/src/main/java/com/imaginaryrhombus/proctimer/application/TimerEndNotification.kt

```diff
@@ -0,0 +1,87 @@
+package com.imaginaryrhombus.proctimer.application
+
+import android.app.NotificationChannel
+import android.app.NotificationManager
+import android.app.PendingIntent
+import android.content.Context
+import android.content.Intent
+import android.os.Build
+import android.os.Message
+import androidx.core.app.NotificationCompat
+import androidx.lifecycle.LiveData
+import com.imaginaryrhombus.proctimer.R
+import com.imaginaryrhombus.proctimer.TimerActivity
+
+/**
+ * タイマーが動作終了した後の通知.
+ */
+class TimerEndNotification(context: Context) {
+
+    /**
+     * 通知に使用するID.
+     */
+    private val notificationId = "timer_notification_id"
+
+    /**
+     * 通知に使用する番号.
+     */
+    private val notificationTagId = 98
+
+    /**
+     * 通知管理のマネージャー.
+     */
+    private val notificationManager =
+        context.getSystemService(Context.NOTIFICATION_SERVICE) as NotificationManager
+
+    /**
+     * 互換性の高い通知を作るためのビルダー.
+     */
+    private val notificationCompatBuilder =
+        NotificationCompat.Builder(context, notificationId).apply {
+            setContentTitle(context.getString(R.string.app_name))
+            setContentText("")
+            setSmallIcon(R.mipmap.ic_launcher)
+            setAutoCancel(true)
+
+            val intent = Intent(context, TimerActivity::class.java)
+            intent.flags = Intent.FLAG_ACTIVITY_CLEAR_TOP or Intent.FLAG_ACTIVITY_SINGLE_TOP
+            val pendingIntent = PendingIntent
+                .getActivity(context, 0, intent, 0)
+
+            setContentIntent(pendingIntent)
+    }
```

Comment:
<table data-meta="generated_by_danger">
  <tbody>
    <tr>
      <td>:warning:</td>
      <td width="100%" data-sticky="false"><span data-href="https://github.com/fizxive/ProcTimer/blob/b31a11cf7f617ce8ac48044c6c809fd38815309c/app/src/main/java/com/imaginaryrhombus/proctimer/application/TimerEndNotification.kt#L52"></span>Unexpected indentation (4) (should be 8)</td>
    </tr>
  </tbody>
</table>